index.js

```diff
@@ -0,0 +1,116 @@
+const lazyloadOptions = {
+  rootElement: null,
+  rootMargin: `0px 0px ${window.innerHeight / 2}px 0px`,
+  treshold: 1,
+  lazyloadClass: 'loaded',
+  nativeLazyloadEnabled: 'loading' in HTMLImageElement.prototype,
+
+  getOptions() {
+    return {
+      root: this.rootElement,
+      rootMargin: this.rootMargin,
+      threshold: this.treshold
+    };
+  }
+};
+
+let lazyObserver = null;
+let resizeListener = null;
+let elements = [];
+let images = [];
+
+const setSource = element => {
+  const { src, srcset, wait } = element.dataset;
+
+  if (src) {
+    element.src = src;
+  }
+
+  if (srcset) {
+    element.srcset = srcset;
+  }
+
+  if (wait) {
+    element.onload = () => element.classList.add(lazyloadOptions.lazyloadClass);
+  }
+
+  element.removeAttribute('data-lazy');
+};
+
+const loadElements = () => {
```

Comment:
Can we try and make functions more... _functional_ ™? I don't like that the functions are changing variables from the upper scope in general. Functions should be able to exist and be called without the need for the surrounding code.